legate/core/store.py

```diff
@@ -1038,6 +1131,46 @@ def __repr__(self) -> str:
     # Convert a store in N-D space to that in (N+1)-D space.
     # The extra_dim specifies the added dimension
     def promote(self, extra_dim: int, dim_size: int = 1) -> Store:
+        """
+        Adds an extra dimension to the store. Value of ``extra_dim`` decides
+        where a new dimension should be added, and each dimension `i`, where
+        `i` >= ``extra_dim``, is mapped to dimension `i+1` in a returned store.
+        A returned store provides a view to the input store where the values
+        are broadcasted along the new dimension.
+
+        For example, for a 1D store ``A`` contains ``[1, 2, 3]``,
+        ``A.promote(0, 2)`` yields:
```

Comment:
might want to say "yields an array equivalent to"

Not specific to this PR, but the semantics here are confusing to me.  If I promote an array, I have a view which means modifying the promoted store would modify the original values.  But `A[0][0] = 1; A[0][1] = 5` are both views onto the same value?

Reply:
> might want to say "yields an array equivalent to"

Funny that I actually wrote like that first and shortened it, as I thought the reader wouldn't notice the subtlety. let me put that back.

> If I promote an array, I have a view which means modifying the promoted store would modify the original values. But A[0][0] = 1; A[0][1] = 5 are both views onto the same value?

aliased updates have undefined semantics. what value the original store will get depends on the evaluation order. that's something I should clarify in a follow-on programming model doc. (we have a similar issue with indirect copies.)